internal/server/template.go

```diff
@@ -51,16 +50,16 @@ func ServeTemplatedFile(publicDir, file, assetPath, duoSelfEnrollment, rememberM
 		var scheme = "https"
 
 		if !https {
-			proto := string(ctx.Request.Header.Peek(fasthttp.HeaderXForwardedProto))
+			proto := string(ctx.XForwardedProto())
 			switch proto {
 			case "":
-				scheme = "http"
-			default:
+				break
+			case "http", "https":
 				scheme = proto
 			}
 		}
 
-		baseURL := scheme + "://" + string(ctx.Request.Host()) + base + "/"
+		baseURL := scheme + "://" + string(ctx.XForwardedHost()) + ctx.Configuration.Server.Path + "/"
```

Comment:
```suggestion
		baseURL := scheme + "://" + string(ctx.XForwardedHost()) + base + "/"
```

This needs to change otherwise it'll always include the path.